File: typescript/packages/bench/src/rules/deviceAgeRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class DeviceAgeRule implements RuleContract {
    readonly key = "device-age";

    evaluate(data: Dataset): boolean {
        return data.devices.length >= 1;
    }
}

File: rust/crates/atelier-bench/src/rules/device_age_rule.rs
//! Rule 9/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct DeviceAgeRule;

impl RuleContract for DeviceAgeRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        !data.devices.is_empty()
    }
}
